src/mpid/ch4/src/ch4_spawn.c

```diff
@@ -68,10 +69,22 @@ int MPID_Comm_spawn_multiple(int count, char *commands[], char **argvs[], const
         }
     }
 
+    int bcast_ints[2];
+    if (comm_ptr->rank == root) {
+        bcast_ints[0] = total_num_processes;
+        bcast_ints[1] = spawn_error;
+    }
     MPIR_Errflag_t errflag = MPIR_ERR_NONE;
-
-    mpi_errno = MPIR_Bcast(&spawn_error, 1, MPI_INT, root, comm_ptr, &errflag);
+    mpi_errno = MPIR_Bcast(bcast_ints, 2, MPI_INT, root, comm_ptr, &errflag);
     MPIR_ERR_CHECK(mpi_errno);
+    if (comm_ptr->rank != root) {
+        total_num_processes = bcast_ints[0];
+        spawn_error = bcast_ints[1];
+        MPIR_Assert(total_num_processes > 0 && total_num_processes < 10);
```

Comment:
Why less than 10?

Reply:
Oops, it is a leftover from debugging. I will delete this line. Good catch!